confgenerator/confgenerator_test.go

```diff
@@ -114,6 +83,52 @@ func testPlatformGenerateConfTests(t *testing.T, platform platformConfig) {
 		err = testGenerateConf(t, platform, testDir)
 		assert.NilError(t, err)
 	})
+
+	t.Run("valid", func(t *testing.T) {
+		t.Parallel()
+
+		runTestsInDir(
+			t,
+			platform,
+			validTestdataDirName,
+			func(t *testing.T, err error, _ string) {
+				assert.NilError(t, err)
+			},
+		)
+	})
+
+	t.Run("invalid", func(t *testing.T) {
+		t.Parallel()
+
+		runTestsInDir(
+			t,
+			platform,
+			invalidTestdataDirName,
+			func(t *testing.T, err error, testDir string) {
+				assert.Assert(t, err != nil, "expected test config to be invalid, but was successful")
+				goldenErrorPath := filepath.Join(testDir, errorGolden)
+				golden.Assert(t, err.Error(), goldenErrorPath)
+			},
+		)
+	})
+}
+
+func runTestsInDir(
+	t *testing.T,
+	platform platformConfig,
+	testTypeDir string,
+	errAssertion func(*testing.T, error, string),
+) {
+	platformTestDir := filepath.Join(testTypeDir, platform.OS)
+	testNames := getTestsInDir(t, platformTestDir)
+
+	for _, testName := range testNames {
+		t.Run(testName, func(t *testing.T) {
+			testDir := filepath.Join(platformTestDir, testName)
```

Comment:
Maybe `t.Parallel()`?

Reply:
Added in [e3d7607](https://github.com/GoogleCloudPlatform/ops-agent/pull/811/commits/e3d7607aea8abf17cea4eb14c04ff3d2a8fcd85b)